content/en/blog/2020/on-demand-egress-gateways/index.md

```diff
@@ -0,0 +1,500 @@
+---
+title: On Demand Egress Gateways
+subtitle: Deploy Egress Gateways independently in a seamless way
+description: Deploying Egress Gateways independently to have fine-grained control of egress communication.
+publishdate: 2020-11-07
+attribution: Antonio Berben (At Deutsche Telekom - PAN-NET)
+keywords: [configuration,egress,gateway,external,service]
+target_release: 1.8.0
+---
+
+At [`Deutsche Telekom::PAN-NET`](https://pan-net.cloud/aboutus), we have embraced `Istio` as the umbrella to cover our services.
+
+Unfortunately, there are services which have not been migrated to `kubernetes` yet or, simply, cannot be. Therefore, those ones have to remain as external services.
+
+## Scenario
+
+If you are familiar with Istio, one of the offered ways to connect to external services is through [Egress Gateway](/docs/tasks/traffic-management/egress/egress-gateway/).
+
+However, if you want to satisfy the [Single-responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle), you will need to deploy multiple and individual (1..N) `Egress Gateways` as the picture shows.
+
+{{< image width="75%" ratio="45.34%"
+    link="./on-demand-egress-gateway-overview.svg"
+    alt="Overview multiple Egress Gateways"
+    caption="Overview multiple Egress Gateways"
+    >}}
+
+With this model, one `Egress Gateway` is in charge of only one upstream service.
+
+Although the `IstioOperator` spec allows you to deploy multiple `Egress Gateways`, the manifest can become unmanageable.
+
+{{< text yaml >}}
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+[...]
+spec:
+    egressGateways:
+    - name: egressgateway-1
+      enabled: true
+    - name: egressgateway-2
+      enabled: true
+    [egressgateway-3, egressgateway-4 ,...]
+    - name: egressgateway-N
+      enabled: true
+[...]
+{{< /text >}}
+
+As a benefit of decoupling Egress Getaways from IstioOperator manifest, you have enabled the possibility of setting up **custom readiness probes** to have both services (gateway and upstream service) aligned.
+
+As well, you can apply the sidecar pattern to inject, for example, OPA into the pod to perform authorization with complex rules ([OPA envoy plugin](https://github.com/open-policy-agent/opa-envoy-plugin)).
+
+{{< image width="75%" ratio="45.34%"
+    link="./on-demand-egress-gateway-authz.svg"
+    alt="Authorization with OPA and `healthcheck` to external service"
+    caption="Authorization with OPA and `healthcheck` to external"
+    >}}
+
+At this point, you might be convinced that this is the right path to go.
+
+## Solution
+
+There are several ways to achieve this task:
```

Comment:
you generally achieve a _solution/outcome,_ or _perform_ a task

Reply:
:sweat_smile: . Done!